### PR TITLE
enhancement(observability): Add registered `BytesSent` event

### PR DESCRIPTION
**Pull request by bruceg**
This adds support for registering the existing `BytesSent` event and puts it
into use in all sinks except for the new-style stream sinks. That change is
blocked by a complication in how the driver handles emitting the `BytesSent`
event centrally.

### CODE REVIEW

**Review by lukesteensen — APPROVED** [review]

File: src/sinks/mod.rs [hunk 1]
```diff
@@ -123,6 +123,7 @@ pub enum HealthcheckError {
 #[configurable_component]
 #[allow(clippy::large_enum_variant)]
 #[derive(Clone, Debug)]
+#[allow(clippy::large_enum_variant)]
```
Comment (tobz) [inline src/sinks/mod.rs:126]: Duplicate of the same thing two lines up.. :P

**Review by tobz — APPROVED** [review]
All looks fairly straightforward to me.  Found one little duplicate line but other than that, all good. 👍🏻

**Review by jszwedko — APPROVED** [review]
👍 seems pretty straightforward to me given the other registered events.

File: src/sinks/mod.rs (hunk 1 above)
Comment (bruceg) [reply src/sinks/mod.rs:126]: Hah, merge glitch, will fix.

**Pull request merged**